### PR TITLE
Ci

### PR DESCRIPTION
**Pull request by mh-cbon**
Added CI and appropriate badges, i hope it help people to have confidence that it worth to give a chance to the lib.
Also windows stranger should be able to find some hints using the appveyor file.

Note that you ll have to signin then enable the repo in each ci services so the badges work.

### CODE REVIEW

**Comment by christian-roggia** [comment]
See https://github.com/anacrolix/torrent/issues/171#issuecomment-328379523

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+
```
Comment (anacrolix) [inline .travis.yml]: I have CircleCI doing linux builds, you may not need travis. If that's the case, let's remove this.

File: README.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
+[![travis Status](https://travis-ci.org/anacrolix/go-libutp.svg?branch=master)](https://travis-ci.org/anacrolix/go-libutp)
```
Comment (anacrolix) [inline README.md]: Use the CircleCI badge?

File: conn.go [hunk 3]
```diff
@@ -1,5 +1,10 @@
+//+build !windows
```
Comment (anacrolix) [inline conn.go]: I think @christian-roggia advice here has been taken a little too literally. Use whatever code works on both unix and windows targets, I suspect your sizeof(WhateverAny) will work on both.

File: conn_windows.go [hunk 4]
```diff
@@ -0,0 +1,277 @@
+//+build windows
```
Comment (anacrolix) [inline conn_windows.go]: We shouldn't need another file.

File: sockaddr.go [hunk 5]
```diff
@@ -33,7 +33,7 @@ func toSockaddrInet(ip net.IP, port int, zone string) (*C.struct_sockaddr, C.soc
 	if n := copy(rsa.Addr[:], ip); n != 16 {
 		panic(n)
 	}
-	return (*C.struct_sockaddr)(unsafe.Pointer(&rsa)), C.socklen_t(syscall.SizeofSockaddrInet6)
+	return (*C.struct_sockaddr)(unsafe.Pointer(&rsa)), C.socklen_t(unsafe.Sizeof(rsa))
```
Comment (anacrolix) [inline sockaddr.go]: If this strategy works on both unix and windows, lets stick with it.

**Comment by anacrolix** [comment]
I've taken the first commit in this PR into the windows branch.

**Comment by mh-cbon** [comment]
@anacrolix i feel stupid to ask about that, which circle ci file ?
I m fine anything so a badge can be put, i d just say, now its ready, just got to click sign in and enable repos :=)

File: sockaddr.go (hunk 5 above)
Comment (mh-cbon) [reply sockaddr.go]: yup, need revert, lets sort out the things about circle. and it will happen.

Comment (anacrolix) [reply sockaddr.go]: https://circleci.com/gh/anacrolix/go-libutp

There's not currently a circle turd for this project as it's simple to test, but I'll add one eventually modelled on the one I have here: https://github.com/anacrolix/torrent/blob/master/.circleci/config.yml

Comment (mh-cbon) [reply sockaddr.go]: works for me. rm windows additions, rm .travis, add circleci. tbd.

**Comment by mh-cbon** [comment]
> I've taken the first commit in this PR into the windows branch.

this pr is useless, you did it in windows branch. 😕

**Comment by anacrolix** [comment]
@mh-cbon, the Windows CI has started failing recently due to an old Go version. Did you want to take a stab at updating it?

**Pull request closed without merging**